Apps/Src/ReadSwitches.c

```diff
@@ -0,0 +1,167 @@
+//#include "Tasks.h"
+#include <os.h>
+#include "Switches.h"
+#include "Tasks.h"
+
+/**
+ * @brief Reads Switches using Switches Driver and signals appropriate
+ * Semaphores when Switches state changes. Also updates a switches global defined in StateTypes.h, declared in Tasks.c and externed in Tasks.h
+*/
+void Task_ReadSwitches (void* p_arg) {
```

Comment:
The logic here makes sense. A recommendation:

Instead of reading the switches grouped on how they're used, read all switches up front and update the `carState` struct. Then, you can go through each of the groupings and just read directly from the `carState` struct. That way, the `carState` values are always properly up to date.

Reply:
Don't the semaphores have to get signaled only if there is a difference between the prior car state and the new car state? If I overwrote the prior car state with the new values at the top, how would I know whether to signal or whether the switch had been on forever?